reVX/least_cost_xmission/aoswt_utilities.py

```diff
@@ -0,0 +1,604 @@
+        combo_data = land_data * land_mult
+        combo_data[~self.land_mask] = os_data[~self.land_mask]
+        combo_data = combo_data.astype(dtype)
+
+        if save_tiff:
+            fname = self.COMBO_LAYER_FNAME.format(layer_name)
+            print(f'Saving combined {layer_name} to {fname}')
+            self._save_tiff(combo_data, fname)
+
+        setattr(self, f'_combo_{layer_name}', combo_data)
+
+        print(f'Writing combined data to "{os_layer}" in {aoswt_h5}')
+        combo_data = combo_data[np.newaxis, ...]
+        with h5py.File(aoswt_h5, 'a') as f:
+            if os_layer in f.keys():
+                dset = f[os_layer]
+                dset[...] = combo_data
+            else:
+                f.create_dataset(os_layer, data=combo_data)
+
+    def create_aoswt_h5(self, aoswt_ex_h5, aoswt_h5, overwrite=False):
+        """
+        Create a new h5 file to save AOSWT data in.
+
+        Parameters
+        ----------
+        aoswt_ex_h5 : str
+            Path to existing h5 file w/ off-shore shape
+        aoswt_h5 : str
+            Path for new h5 file to create
+        overwrite : bool, optional
+            Overwrite existing h5 file if True
+
+        """
+        if os.path.exists(aoswt_h5) and not overwrite:
+            raise AttributeError(f'File {aoswt_h5} exits')
+
+        with rex.Resource(aoswt_ex_h5) as res:
+            lats = res['latitude']
+            lngs = res['longitude']
+            global_attrs = res.global_attrs
+
+        assert lats.shape == self._os_shape
+        regions = np.ones(self._os_shape, dtype='uint8')
+
+        with h5py.File(aoswt_h5, 'w') as f:
+            f.create_dataset('longitude', data=lngs)
+            f.create_dataset('latitude', data=lats)
+            f.create_dataset('ISO_regions', data=regions[np.newaxis, ...],
+                             dtype='uint8')
+            for k, v in global_attrs.items():
+                f.attrs[k] = v
+
+    @property
+    def land_mask(self):
+        if self._land_mask is None:
+            cls_name = self.__class__.__name__
+            msg = (f'Must run {cls_name}.create_land_mask() or {cls_name}.'
+                   'load_land_mask() first')
+            raise RuntimeError(msg)
+
+        return self._land_mask
+
+    def profile(self, dtype=None):
+        prof = self._os_profile.copy()
+        if dtype:
+            prof['dtype'] = dtype
+        return prof
+
+    @staticmethod
+    def _extract_profile(template_f):
+        with rio.open(template_f) as ras:
+            profile = {
+                'crs': ras.crs,
+                'transform': ras.transform,
+                'height': ras.height,
+                'width': ras.width,
+                # 'dtype': ras.dtype,
+                'count': 1,
+                'compress': 'lzw'
+            }
+        return profile
+
+    def _load_layer(self, f, val, verbose=False):
+        """
+        Load a layer from a tiff and set appropriate cells to 1
```

Comment:
Can you only handle tiff inputs? Should we support our collection of h5 exclusion layers?

Reply:
Does not seem to be major demand for this request, so I propose we shelve it. In the immediate future, anyone looking to use this function with an h5 exclusion layer can use the [`$ exclusions layers-from-h5`](https://github.com/NREL/reVX/blob/3a3694a171ab672b35865a27ba2f64c470280dcc/reVX/cli.py#L212) command line utility to extract h5 layers as tiff.

Reply:
A little context: I'm not very good at using h5s and I frequently load raster and vector data into QGIS for using basemaps and zooming. QGIS doesn't speak h5. Tiffs are also much smaller than h5 files and easier to copy to and from eagle. So that's why some of this code is setup for using tiffs. I believe all the input files have already been converted to h5 files. Switching over to a pure h5 workflow should not be an issue.